### PR TITLE
[quant][fx][bug] Fix error in convert step for QAT

### PR DESCRIPTION
**Pull request by supriyar**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#45050 [quant][fx][bug] Fix error in convert step for QAT**

Summary:
Update tests to actually test for QAT

Test Plan:
python test/test_quantization.py TestQuantizeFxOps.test_linear

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23808022](https://our.internmc.facebook.com/intern/diff/D23808022)

### CODE REVIEW

File: torch/quantization/fx/quantize.py [hunk 1]
```diff
@@ -511,7 +511,7 @@ def load_arg(a):
 
         to_be_removed = []
         for name, module in model.named_modules():
-            if is_activation_post_process(module):
+            if 'activation_post_process_' in name.split('.')[-1]:
```
Comment (jerryzh168) [inline torch/quantization/fx/quantize.py:514]: why is this changed?

Comment (supriyar) [reply torch/quantization/fx/quantize.py]: This is for the observer modules that are an attr of the fakequant module. If we just check the module type then it returns true and then fails when we try to call `delattr` below.

Comment (jerryzh168) [reply torch/quantization/fx/quantize.py]: I see, maybe make more sense to fix this by avoiding checking the submodule of fakequantize module?

Comment (supriyar) [reply torch/quantization/fx/quantize.py]: The submodule (i.e observer) shows up in `model.named_modules()` as well. So we need some additional check to not add that in the removed list.

Comment (jerryzh168) [reply torch/quantization/fx/quantize.py]: yes, I think so, feel free to move this to a function

File: torch/quantization/fx/quantize.py [hunk 2]
```diff
@@ -126,6 +126,12 @@ def is_activation_post_process(module):
     return (isinstance(module, torch.quantization.ObserverBase) or
             isinstance(module, torch.quantization.FakeQuantize))
 
+def is_submodule_of_fake_quant(name, module, named_modules):
+    parent_name = name.rsplit('.', 1)
+    if len(parent_name) > 1:
+        return is_activation_post_process(named_modules[parent_name[0]])
```
Comment (jerryzh168) [inline torch/quantization/fx/quantize.py:132]: nit: we have a _parent_name helper function I think, you can use that

**Review by jerryzh168 — APPROVED** [review]
Looks good, please use _parent_name
